src/main/java/de/taimos/pipeline/aws/WithAWSStep.java

```diff
@@ -194,7 +204,8 @@ private void withRole(@Nonnull EnvVars localEnv) {
 				
 				String roleARN = String.format("arn:aws:iam::%s:role/%s", accountId, this.step.getRole());
 				
-				AssumeRoleRequest request = new AssumeRoleRequest().withRoleArn(roleARN).withRoleSessionName("Jenkins-" + System.currentTimeMillis());
+				AssumeRoleRequest request = new AssumeRoleRequest().withRoleArn(roleARN).withRoleSessionName("Jenkins-" + System.currentTimeMillis()).withExternalId(this.step.getExternalId());
```

Comment:
Are you sure you can call withExternalId when it is null? Or should it only be called when not null?